src/onegov/search/integration.py

```diff
@@ -375,24 +401,30 @@ def reindex_model(model):
                     q = q.filter(i.polymorphic_on == i.polymorphic_identity)
 
                 for obj in q:
-                    self.es_orm_events.index(self.schema, obj)
+                    self.es_orm_events.index(schema, obj)
+
+            except Exception as e:
+                print(f'Error psql indexing model \'{model}\': {e}')
             finally:
                 session.invalidate()
                 session.bind.dispose()
 
-        # by loading models in threads we can speed up the whole process
+        models = [model for base
+                  in self.session_manager.bases  # type: ignore[attr-defined]
+                  for model in searchable_sqlalchemy_models(base)]
+        index_log.info(f'Number of models to be indexed: {len(models)}')
+
         with ThreadPoolExecutor() as executor:
             results = executor.map(
-                reindex_model, (
-                    model
-                    for base in self.session_manager.bases
-                    for model in searchable_sqlalchemy_models(base)
-                )
+                reindex_model, (model for model in models)
             )
             if fail:
-                tuple(results)
+                print(tuple(results))
 
         self.es_indexer.bulk_process()
+        self.psql_indexer.process()
```

Comment:
Eventually we may want to implement `bulk_process` for the `psql_indexer` in order to improve performance of the complete re-index, if you feel like it you can take a stab at it right now:

You would gather all the `index` tasks grouped by model (you can use `task['type_name']` as the groupby key, they're already sorted by model, because that's how we generated the tasks in the first place) and use `executemany` instead of `execute` to reindex all the relevant rows at once. You'd probably have to refactor `index` slightly, so the part that generates the query can be reused, you'd need to turn `id` into a substitution value like `language` and `data` so the three values can be passed as a list of dictionaries to `executemany`.

Reply:
ok, will try that later